Automattic/wp-calypso
# Marketplace: Use TypeScript plugin selectors in Marketplace

## Proposed Changes

As a part of the work to improve the performance of the plugins selectors which began in https://github.com/Automattic/wp-calypso/pull/72363, the plugins selectors were changed to be in TypeScript. This PR switches the Marketplace code to use the TypeScript versions of the selectors in preparation for the removal of the JavaScript ones.

This is a second attempt at what https://github.com/Automattic/wp-calypso/pull/76788 was doing. What has changed is the return type of `getStatusForPlugin` was specified in https://github.com/Automattic/wp-calypso/pull/77575, which now properly causes a compile error when `pluginInstallStatus.error` is accessed. This access was changed to `pluginInstallStatus?.error`.

## Testing Instructions

Plugin installation:
1. Switch to an site with a plan that supports plugins.
2. Visit `/plugins/:site`.
3. Go through the flow to purchase a paid plan, and verify that it is installed and activated at the end of the flow.
4. Return to `/plugins/:site` and go through the flow to install a free plugin. There is a section called "The free essentials" which contains only free plugins. Verify that it is installed and activated at the end of the flow.

Marketplace test:
1. Visit `/marketplace/test/:site` and verify that the display is identical to what is seen in production, and that there are no console errors.



## Pre-merge Checklist
